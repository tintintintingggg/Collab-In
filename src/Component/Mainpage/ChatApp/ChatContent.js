import React from 'react';
import {justifyNumberToTwoDigits} from '../lib';
import {db} from '../../../utils/firebase';

class ChatContent extends React.Component{
    constructor(props){
        super(props);
        this.chatContent=React.createRef();
        this.state={
            content: [],
            isLoading: false
        }
    }
    formatDate(time){
        let date = new Date(time).getDate();
        date = justifyNumberToTwoDigits(date);
        let month = new Date(time).toLocaleString('en-us', {month: 'long'});
        let year = new Date(time).getFullYear();
        return {date: date, month: month, year: year};
    }
    showDate(showDataarea, time){
        let data = this.formatDate(time);
        showDataarea = <div className="date-separator" key={time}>{`${data.date}  ${data.month}  ${data.year}`}</div>;
        return showDataarea;
    }
    formatTime(time){
        let hour = new Date(time).getHours();
        hour = justifyNumberToTwoDigits(hour);
        let minute = new Date(time).getMinutes();
        minute = justifyNumberToTwoDigits(minute);
        return hour+':'+minute;
    }
    render(){
        let content = [];
        if(this.state.content.length>0){
            let date;
            this.state.content.forEach((message, index)=>{
                let item;
                let dateSeparator;
                if(!date || date !== new Date(message.time).getDate()){
                    content.push(this.showDate.call(this, dateSeparator, message.time))
                }
                date = new Date(message.time).getDate();
                if(this.props.currentUser.uid !== message.user){
                   let photoSrc = message.photo ? message.photo : '/images/user-1.png';
                    item = <div key={index} className="message-item other-user">
                        <div className="user-pic"><img src={photoSrc} /></div>
                        <div className="message-content">
                            <div className="content-name">{message.name}</div>
                            <div className="text-wrap">
                                <div className="content-text">{message.text}</div>
                                <div className="content-time">
                                    <div><img src="/images/clock.png" /></div>
                                    {this.formatTime.call(this, message.time)}
                                </div>
                            </div>
                        </div>
                    </div>;
                }else{
                    item = <div key={index} className="message-item myself">
                        <div className="message-content">
                            <div className="text-wrap">
                                <div className="content-text">{message.text}</div>
                                <div className="content-time">
                                    <div><img src="/images/clock.png" /></div>
                                    {this.formatTime.call(this, message.time)}
                                </div>
                            </div>
                        </div>
                    </div>;
                }
                content.push(item);
            })
        }
        if(this.chatContent.current){
            this.chatContent.current.scrollIntoView(false);
            this.chatContent.current.scrollTop = 9999999999;
        }
        return <div className="chat-content" ref={this.chatContent}>
           {content}
        </div>
    }
    componentDidMount(){
        this.unsubscribe = db.collection('chatrooms').doc(this.props.docId).collection('messages')
        .orderBy("time")
        .onSnapshot((snap)=>{
            let arr = [];
            snap.docChanges().forEach(doc=>{
                if(doc.type === 'added'){
                    arr.push(doc.doc.data())
                }
            })
            this.setState(prevState => ({
                isLoading: true,
                content: prevState.content.concat(arr)
            }))
        })
        let div = this.chatContent.current;
        div.scrollTop = (div.scrollHeight)-(div.clientHeight);
    }
    componentWillUnmount(){
        this.unsubscribe();
    }
}

export {ChatContent};