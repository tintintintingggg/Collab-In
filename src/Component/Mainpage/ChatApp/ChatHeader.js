import React from 'react';
import {db} from '../../../utils/firebase';

class ChatHeader extends React.Component{
    constructor(props){
        super(props);
        this.state={
            members: [],
            showMembers: false
        }
    }
    showMembers(){
        this.setState(prevState=>({
            showMembers: !prevState.showMembers
        }));
    }
    render(){
        let list = '';
        let membersCount = ' ... ';
        if(this.state.members.length>0){
            let arr=[];
            this.state.members.map(doc=>{
                let item = <div key={doc}>{doc}</div>
                arr.push(item);
            })
            list = arr;
            membersCount = this.state.members.length;
        }
        return <div className="chat-header">
                <div>
                    <div className="shrink-btn" onClick={this.props.handleChatRoom}><img src="/images/minimize.png" /></div>
                    <div className="group-members" onClick={this.showMembers.bind(this)}>Members ({membersCount})</div>
                </div>
                <div className="members-list" style={{display: this.state.showMembers ? 'flex' : 'none'}}
                >{list}</div>
        </div>;
    }
    componentDidMount(){
        this.unsubscribe = db.collection('chatrooms').doc(this.props.docId).collection('members')
        .onSnapshot((snap)=>{
            snap.docChanges().forEach(doc=>{
                if(doc.type === 'added'){
                    db.collection('users').doc(doc.doc.data().id).get()
                    .then(data=>{
                        this.setState(prevState=>({
                            members: prevState.members.concat([data.data().name])
                        }))
                    })
                    .catch((error)=>{console.log(error.message)})
                }
            })
        })
    }
    componentWillUnmount(){
        this.unsubscribe();
    }
}

export default ChatHeader;