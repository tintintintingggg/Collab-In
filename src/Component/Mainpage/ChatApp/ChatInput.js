import React from "react";
import {db} from "../../../utils/firebase";
// redux
import {connect} from "react-redux";

class EmojiBtn extends React.Component{
  constructor(props){
    super(props);
  }
  render(){
    return <button onClick={this.props.handleEmojisToShow}>
      <img src="/images/chat-input-add.png" />
    </button>;
  }
}
class EmojiList extends React.Component{
  constructor(props){
    super(props);
    this.state={
      emojiList: [
        "&#128512;", "&#128513;", "&#128514;","&#129315;", "&#128515;", 
        "&#128516;", "&#128517;", "&#128518;", "&#128521;", "&#128522;",	
        "&#128523;", "&#128526;", "&#128525;", "&#128536;",	"&#128535;",	
        "&#128537;", "&#128538;", "&#128578;", "&#129303;",	"&#129300;",	
        "&#128528;", "&#128529;", "&#128566;", "&#128580;", "&#128527;",	
        "&#128547;", "&#128549;", "&#128558;", "&#129296;", "&#128559;",
        "&#128554;", "&#128555;", "&#128564;", "&#128524;", "&#129299;",	
        "&#128539;", "&#128540;", "&#128541;", "&#129316;", "&#128530;",	
        "&#128531;", "&#128532;", "&#128533;", "&#128579;",	"&#129297;",
        "&#128562;", "&#128577;", "&#128534;", "&#128542;",	"&#128543;",	
        "&#128548;", "&#128546;", "&#128557;", "&#128550;",	"&#128551;",	
        "&#128552;", "&#128553;", "&#128556;", "&#128560;",	"&#128561;",	
        "&#128563;", "&#128565;", "&#128545;", "&#128544;", "&#128519;",	
        "&#129312;", "&#129313;", "&#129317;", "&#128567;",	"&#129298;",	
        "&#129301;", "&#129314;", "&#129319;", "&#128520;",	"&#128127;",	
        "&#128121;", "&#128122;", "&#128128;", "&#128123;",	"&#128125;",	
        "&#128126;", "&#129302;", "&#128169;", "&#128570;",	"&#128568;",	
        "&#128569;", "&#128571;", "&#128572;", "&#128573;",	"&#128576;",	
        "&#128575;", "&#128574;", "&#128584;", "&#128585;", "&#128586;",
        "&#128129;", "&#127995;", "&#127996;", "&#127997;", "&#127998;",	
        "&#127999;", "&#128587;", "&#127995;", "&#127996;", "&#127997;",	
        "&#127998;", "&#127999;",		
      ]
    };
  }
  pickEmoji(e){
    this.props.handleInputPlusEmoji(e.target.textContent);
    this.props.handleEmojisToShow();
  }
  render(){
    let emojiList = "";
    if(this.props.showEmojis){
      emojiList = [];
      for(let i = 0; i<this.state.emojiList.length; i++){
        let item = <span 
          dangerouslySetInnerHTML={{__html: this.state.emojiList[i]+" "}} 
          key={i}
          className="my-emoji"
          onClick={this.pickEmoji.bind(this)}
        ></span>;
        emojiList.push(item);
      }
    }
    return <div className="emojis" style={{display: this.props.showEmojis ? "block" : "none"}}>
      {emojiList}
    </div>;
  }
}

class ChatInput extends React.Component{
  constructor(props){
    super(props);
    this.input = React.createRef();
    this.state={
      input: null,
      showEmojis: false
    };
  }
  getInput(e){
    this.setState({
      input: e.target.value
    });
  }
  sendInput(e){
    e.preventDefault();
    if(this.props.user){
      if(this.state.input){
        this.input.current.value = null;
        db.collection("chatrooms").doc(this.props.docId).collection("messages").doc().set({
          user: this.props.user.uid,
          name: this.props.user.displayName,
          photo: this.props.user.photoURL,
          text: this.state.input,
          time: Date.now()
        })
          .then(()=>{
            this.setState({
              input: null
            });
          })
          .catch((error)=>{console.log(error.message);});
      }else{
        alert("Type Something!");
      }
    }
  }
  pickEmoji(e){
    this.input.current.value += e.target.textContent;
    this.setState({
      input: this.input.current.value
    });
    this.handleEmojisToShow();
  }
  handleEmojisToShow(){
    this.setState(prevState=>({
      showEmojis: !prevState.showEmojis
    }));
  }
  handleInputPlusEmoji(emoji){
    this.input.current.value += emoji;
    this.setState({
      input: this.input.current.value
    });
  }
  render(){
    return  <div className="chat-input">
      <EmojiList
        showEmojis={this.state.showEmojis}
        handleEmojisToShow={this.handleEmojisToShow.bind(this)}
        handleInputPlusEmoji={this.handleInputPlusEmoji.bind(this)}
      />
      <EmojiBtn
        handleEmojisToShow={this.handleEmojisToShow.bind(this)}
      />
      <form onSubmit={this.sendInput.bind(this)}>
        <input
          id="chat-input-block"
          type="text" 
          placeholder="Say something..." 
          onChange={this.getInput.bind(this)}
          ref={this.input}
        />
        <button id="chat-input-submit" type="submit"><img src="/images/chat-input-send.png" /></button>
      </form>
    </div>;
  }
}

const mapStateToProps = (store)=>{
  return{user: store.userReducer.user};
};
ChatInput = connect(mapStateToProps)(ChatInput);
export {ChatInput, EmojiList};