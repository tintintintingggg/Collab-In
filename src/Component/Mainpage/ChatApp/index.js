import React from "react";
import ChatContent from "./ChatContent";
import ChatHeader from "./ChatHeader";
import {ChatInput} from "./ChatInput";
import "../../../css/ChatApp.css";

class ChatApp extends React.Component{
  constructor(props){
    super(props);
    this.chatapp = React.createRef();
    this.handleDragChatRoom = this.handleDragChatRoom.bind(this);
    this.startDrag = this.startDrag.bind(this);
    this.stopDrag = this.stopDrag.bind(this);
    this.state={
      startX: null,
      startWidth: null
    };
  }
  getBorder(e){
    let div = this.chatapp.current;
    let rect = div.getBoundingClientRect();
    let x = e.clientX - rect.left;
    let thickness = 5;
    div.style.cursor = x<thickness && x>(-thickness) ? "col-resize" : "default";
  }
  dragging(e){
    let startX = this.state.startX;
    let startWidth = this.state.startWidth;
    let div = this.chatapp.current;
    let parentDivRect = div.parentNode.getBoundingClientRect();
    let parentDivWidth = parentDivRect.right-parentDivRect.left;
    let changingWidth = startX-e.clientX;
    let resizedWidth = (startWidth+changingWidth)*100/parentDivWidth;
    if(resizedWidth<70 && resizedWidth>20){
      div.style.width = resizedWidth+"%";
      div.previousSibling.style.width = (100-resizedWidth)+"%";
    }
  }
  handleDragChatRoom(e){
    this.getBorder(e);
    if(this.state.startX){
      this.dragging(e);
    }
  }
  startDrag(e){
    let div = this.chatapp.current;
    let rect = div.getBoundingClientRect();
    let x = e.clientX - rect.left;
    let thickness = 5;
    if(x<thickness && x>(-thickness)){
      this.setState({
        startX: e.clientX,
        startWidth: rect.right-rect.left
      });
    }
  }
  stopDrag(){
    this.setState({
      startX: null
    });
  }
  render(){
    return <div 
      className='chat-app' 
      id="chat-app" 
      style={{display: this.props.chatAppBlockIsOpen ? "flex" : "none"}}
      ref={this.chatapp}
    >
      <ChatHeader
        docId={this.props.docId}
        handleChatRoom={this.props.handleChatRoom}
      />
      <ChatContent
        docId={this.props.docId}
      />
      <ChatInput 
        docId={this.props.docId}
      />
    </div>;
  }
  componentDidMount(){
    window.addEventListener("mousemove", this.handleDragChatRoom);
    window.addEventListener("mousedown", this.startDrag);
    window.addEventListener("mouseup", this.stopDrag);
  }
  componentWillUnmount(){
    window.removeEventListener("mousemove", this.handleDragChatRoom);
    window.removeEventListener("mousedown", this.startDrag);
    window.removeEventListener("mouseup", this.stopDrag);
  }
}

export default ChatApp;