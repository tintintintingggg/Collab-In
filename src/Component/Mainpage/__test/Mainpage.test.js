import {EmojiList} from '../ChatApp/ChatInput';
import React from 'react';
import renderer from 'react-test-renderer';

test('ChatInput Emoji UI Test', () => {
    const props = {
        showEmojis: false
    }
    const tree = renderer.create(<EmojiList {...props} />).toJSON()
    expect(tree).toMatchSnapshot();
})
test('ChatInput Emoji UI Test', () => {
    const props = {
        showEmojis: true
    }
    const tree = renderer.create(<EmojiList {...props} />).toJSON()
    expect(tree).toMatchSnapshot();
})

// test('ChatInput Emoji UI Test', () => {
//     const props = {
//         showEmojis: true
//     }
//     const component = renderer.create(<EmojiList {...props} />)
//     const tree = component.toJSON()
//     // component.root.findAllByType('span')[0].props.onClick();
//     component.root.findAllByType('span').forEach((item)=>{
//         item.props.onClick();
//     });
//     expect(tree).toMatchSnapshot();
// })

