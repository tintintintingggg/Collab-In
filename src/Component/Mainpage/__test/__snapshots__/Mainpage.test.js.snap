// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput Emoji UI Test 1`] = `
<div
  className="emojis"
  style={
    Object {
      "display": "none",
    }
  }
>
  
</div>
`;

exports[`ChatInput Emoji UI Test 2`] = `
<div
  className="emojis"
  style={
    Object {
      "display": "block",
    }
  }
>
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128512; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128513; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128514; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129315; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128515; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128516; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128517; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128518; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128521; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128522; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128523; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128526; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128525; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128536; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128535; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128537; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128538; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128578; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129303; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129300; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128528; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128529; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128566; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128580; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128527; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128547; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128549; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128558; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129296; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128559; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128554; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128555; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128564; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128524; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129299; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128539; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128540; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128541; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129316; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128530; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128531; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128532; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128533; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128579; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129297; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128562; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128577; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128534; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128542; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128543; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128548; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128546; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128557; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128550; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128551; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128552; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128553; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128556; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128560; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128561; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128563; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128565; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128545; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128544; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128519; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129312; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129313; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129317; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128567; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129298; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129301; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129314; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129319; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128520; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128127; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128121; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128122; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128128; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128123; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128125; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128126; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#129302; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128169; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128570; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128568; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128569; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128571; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128572; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128573; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128576; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128575; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128574; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128584; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128585; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128586; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128129; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127995; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127996; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127997; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127998; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127999; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#128587; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127995; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127996; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127997; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127998; ",
      }
    }
    onClick={[Function]}
  />
  <span
    className="my-emoji"
    dangerouslySetInnerHTML={
      Object {
        "__html": "&#127999; ",
      }
    }
    onClick={[Function]}
  />
</div>
`;
